### PR TITLE
Migrate `Team.GetMember` to Sync by default

### LINKED ISSUE #11123: Migrate "Team.GetMember" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16176)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetMember` in the `Team` store.

The expected way to implement it is, go to the `Team` store implementation in the `store/sqlstore/` directory, modify the method `GetMember` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by rvillablanca**
This PR makes Team.GetMember sync now. Fix #11123

### CODE REVIEW

File: app/team.go [hunk 1]
```diff
@@ -502,8 +501,8 @@ func (a *App) joinUserToTeam(team *model.Team, user *model.User) (*model.TeamMem
 		tm.SchemeAdmin = true
 	}
 
-	etmr := <-a.Srv.Store.Team().GetMember(team.Id, user.Id)
-	if etmr.Err != nil {
+	result, err := a.Srv.Store.Team().GetMember(team.Id, user.Id)
```
Comment (jespino) [inline app/team.go]: probably it can be just `rtm, err := ...` and remove the `rtm := result`

File: app/team.go [hunk 2]
```diff
@@ -685,11 +684,11 @@ func (a *App) GetTeamsForUser(userId string) ([]*model.Team, *model.AppError) {
 }
 
 func (a *App) GetTeamMember(teamId, userId string) (*model.TeamMember, *model.AppError) {
-	result := <-a.Srv.Store.Team().GetMember(teamId, userId)
```
Comment (jespino) [inline app/team.go]: Here you can just `return a.Srv.Store.Team().GetMember(teamId, userId)`

File: app/team.go (hunk 1 above)
Comment (rvillablanca) [reply app/team.go]: Can I remove the comment above `rtm := result` ?

File: app/team.go (hunk 2 above)
Comment (rvillablanca) [reply app/team.go]: You are right!

File: app/team.go (hunk 1 above)
Comment (jespino) [reply app/team.go]: Yes, I think so

**Comment by hmhealey** [comment]
Re-running the build since this seems to have hit a random build failure

**Comment by jespino** [comment]
Merged! thanks @rvillablanca!

**Pull request merged**